File: frontend/src/app/modules/trotamundos/files/file.vm.ts
import { StorageVM } from 'app/shared/storage/storage.vm';

export class FileVM {
    id: string = null!; // Guid
    group: string = null!;
    storage: StorageVM = null!;
    displayName: string = null!;
    description: string = null!;
    position: number = 0;

    constructor(init?: Partial<FileVM>) {
        Object.assign(this, init);
    }
}

File: backend/src/trotamundos/lrembecki.core.trotamundos/Files/FileVM.cs
using lrembecki.core.storage;

namespace lrembecki.core.trotamundos.Files;

public record FileVM(
    Guid Id,
    string Group,
    int Position,
    string DisplayName,
    string Description,
    Guid StorageId,
    StorageVM Storage)
{
    internal static FileVM Map(FileEntity entity)
    {
        if (entity == null) return null!;

        return new FileVM(
            entity.Id,
            entity.Group,
            entity.Position,
            entity.DisplayName,
            entity.Description,
            entity.StorageId,
            StorageVM.Map(entity.Storage));
    }
    
}
